### ISSUE TITLE
Bug: invalid characters in filenames are ignored

### ISSUE THREAD

**Issue by Pierstoval** [danmunn/redmine_dmsf#822]
I've been using DMSF for years now, and when I upgraded the plugin to the latest version (1.6.1), I cannot nor manage nor upload nor delete any file that has the `[` or `]` character in it. And I have tons of files like this.

The error I encounter when uploading is this one `warning_some_files_were_not_commited` and when removing it's `warning_some_entries_were_not_deleted`.

I tested with the latest `1.6.1-devel` version, the `master` branch, and the error still occurs.

Is there any way these characters could be allowed again?

**Comment by picman** [comment]
The latest released version is 1.6.0 and I'd recommend to upgrade to stable versions only. However testing of devel versions is  always welcomed. 
Square brackets were added into an invalid characters set in 1.5.6. See #498. So it's not a question of the last release.
As a workaround you can remove '[]' character from the invalid characters set in app/models/dmsf_folder.rb as follows:
```ruby
47 -- INVALID_CHARACTERS = /\A[^\[\]\/\\\?":<>#%\*]*\z/.freeze
47 ++ INVALID_CHARACTERS = /\A[^\/\\\?":<>#%\*]*\z/.freeze
```
and restart the application. But better to not use them in you file names. There is no plan to support them in the near future, I'm afraid.

**Comment by Pierstoval** [comment]
I got the idea, and I'll use this workaround, thanks 👍 

In the meantime, I'll try to find something, a script maybe, to update all files and remove the square brackets 👍
